src/foam/core/FObject.js

```diff
@@ -808,6 +808,12 @@ foam.CLASS({
 
     function compareTo(other) {
       if ( other === this ) return 0;
+
+      // Use comparator of other if it is not an FObject.
+      if ( ! foam.core.FObject.isInstance(other) ) {
```

Comment:
This can be simpler now, right? With the addition of precedence + `FObject` being last, we can use `if (...other not FObj...) return foam.util.compare(this, other);` which will take the right path for us.